README.md

```diff
@@ -36,3 +36,157 @@ This is the shared library defining public classes that constitute the worker in
+    task = task;
+    rootDocument=task.document;
+}
+```
+For more details of the <a name="onProcessTask"></a>`TaskEventObject` refer the java implementation of the class, [TaskEventObject.java](https://github.com/CAFDataProcessing/worker-document/blob/develop/worker-document/src/main/java/com/hpe/caf/worker/document/scripting/events/TaskEventObject.java) 
+
+```
+/* global thisScript */
+
+function onProcessTask(e)
+{	
+    // e.application  (read-only)
+    // e.task         (read-only)
+    // e.rootDocument (read-only)
+}
+```
+
+#### onBeforeProcessDocument
+
+This event will be executed after `onProcessTask` and before processing of a document. 
+
+This function is passed `CancelableDocumentEventObject`  with `Document` as an argument.
+
+An example of the `CancelableDocumentEventObject` is below.
+```
+CancelableDocumentEventObject(Document document)
+{
+    super(document);
+    cancel=false;
+}
+```
+For more details of the `CancelableDocumentEventObject`,  refer the Java implementation of for the class,
+[CancelableDocumentEventObject.java](https://github.com/CAFDataProcessing/worker-document/blob/develop/worker-document/src/main/java/com/hpe/caf/worker/document/scripting/events/CancelableDocumentEventObject.java) 
+
+```
+function onBeforeProcessDocument(e)
+{
+    // e.application  (read-only)
+    // e.task         (read-only)
+    // e.rootDocument (read-only)
+    // e.document     (read-only)
+    // e.cancel       (writable)  (default: false)
+}
+```
+Output of this function would be the boolean value of event's cancel flag.
+This flag is used to determine if that individual document should be processed by the worker.
+
+#### <a name="onProcessDocument"></a>onProcessDocument
+
+This function will be called to process a document. 
+This function is passed `DocumentEventObject` with `Document` as an argument.
+The `onBeforeProcessDocument`  event has already been triggered to check for cancellation flag.
+
+An example of the `DocumentEventObject` is below.
+```
+DocumentEventObject(Document document){
+    document = document;
+}
+```
+For more details  of the <a name="onProcessDocument"></a>`DocumentEventObject`, refer to the java implementation for the class,  [DocumentEventObject](https://github.com/CAFDataProcessing/worker-document/blob/develop/worker-document/src/main/java/com/hpe/caf/worker/document/scripting/events/DocumentEventObject.java)
+```
+function onProcessDocument(e)
+{
+    // e.application  (read-only)
+    // e.task         (read-only)
+    // e.rootDocument (read-only)
+    // e.document     (read-only)
+}
+```
+This function calls the customization scripts, and if none of them have set the cancellation flag then calls the implementation's `DocumentWorker#processDocument processDocument()` function. 
+
+#### onAfterProcessDocument
+
+This function will be called once the processing of the document completed successfully.
+This function is passed `DocumentEventObject` with `Document` as an argument.
```

Comment:
with `Document` as an argument

Document is not an argument, it is a field on the argument.